### ISSUE TITLE
FR: configure log levels via command-line flag

### ISSUE THREAD

**Issue by DazWilkin** [deislabs/krustlet#237]
I've been educating myself on Krustlet by running e.g. `krustlet-wasi` in the foreground.

The process appears to only log errors but I'd find it useful to review more extensive (e.g. debug) logging.

Please consider adding a flag to both `krustlet-wasi` and `krustlet-wascc` e.g. `--log-level=debug`

**Comment by bacongobbler** [comment]
This is currently possible with the RUST_LOG environment variable. Let us know if that will work for you.

**Comment by thomastaylor312** [comment]
Yeah, we've had conversations about this. It is bad right now (you have to set `RUST_LOG` and the syntax is a little...non-traditional) so I want to make it easier.

@bacongobbler We probably do want something like this. Or another variable besides `RUST_LOG` that takes gross syntax like this: `wascc_host=debug,wascc_provider=debug,wasi_provider=debug,main=debug`. That requires people to know the name of crates and other things. We should just expose a simple log level that users can set

**Comment by DazWilkin** [comment]
Thank you both... I didn't think to try `RUST_LOG` doh. Will try it (tomorrow).

**Comment by bacongobbler** [comment]
I fully agree @thomastaylor312. RUST_LOG is available, but it's certainly wonky. That being said, we do get deeper introspective logging from waSCC and wasmtime for free with RUST_LOG. If we can tie RUST_LOG into a `--log-level` flag, that would be the best of both worlds :)

**Comment by thomastaylor312** [comment]
Yeah, I think that would be the ideal

**Comment by kflansburg** [comment]
What ever we decide to do here, I want to keep `RUST_LOG`. It may be confusing but it is widely used and understood by Rust developers. Could we write a small piece of code that takes something coarse like `LOG_LEVEL=DEBUG` and configures `env_logger` accordingly for all of the crates in the workspace?

**Comment by thomastaylor312** [comment]
> Could we write a small piece of code that takes something coarse like LOG_LEVEL=DEBUG and configures env_logger accordingly for all of the crates in the workspace?

This is exactly what I was imagining @kflansburg. We can keep the environment variable around for advanced usage, but I don't want to require users to have knowledge of Rust or how modules work in order to use this. Right now, if I didn't know what I was doing, I'd set `RUST_LOG=debug` and then have oodles of information show up from underlying things like reqwest, which wouldn't necessarily be desired
